R/backend-.R

```diff
@@ -197,31 +193,62 @@ base_scalar <- sql_translator(
 
   `%>%` = `%>%`,
 
+  substr = sql_substr("SUBSTR"),
+
   # stringr functions
   # SQL Syntax reference links:
   #   MySQL https://dev.mysql.com/doc/refman/5.7/en/string-functions.html
   #   Hive: https://cwiki.apache.org/confluence/display/Hive/LanguageManual+UDF#LanguageManualUDF-StringFunctions
   #   Impala: https://www.cloudera.com/documentation/enterprise/5-9-x/topics/impala_string_functions.html
   #   PostgreSQL: https://www.postgresql.org/docs/9.1/static/functions-string.html
   #   MS SQL: https://docs.microsoft.com/en-us/sql/t-sql/functions/string-functions-transact-sql
-  #   Oracle: https://docs.oracle.com/middleware/bidv1221/desktop/BIDVD/GUID-BBA975C7-B2C5-4C94-A007-28775680F6A5.htm#BILUG685
-  str_length      = sql_prefix("LENGTH"),
-  str_to_upper    = sql_prefix("UPPER"),
-  str_to_lower    = sql_prefix("LOWER"),
+  #   Oracle: https://docs.oracle.com/database/121/SQLRF/functions002.htm#SQLRF51180
+  str_length      = sql_prefix("LENGTH", 1),
+  str_to_upper    = sql_prefix("UPPER", 1),
+  str_to_lower    = sql_prefix("LOWER", 1),
+  str_to_title    = sql_prefix("INITCAP", 1),
   str_replace_all = function(string, pattern, replacement) {
-                      sql_expr(REPLACE(!!string, !!pattern, !!replacement))
-                    },
-  str_detect      = function(string, pattern) {
-                      sql_expr(INSTR(!!string, !!pattern) > 0L)
-                    },
-  str_trim        = function(string, side = c("both", "left", "right")) {
-                      side <- match.arg(side)
-                      switch(side,
-                        left = sql_expr(LTRIM(!!string)),
-                        right = sql_expr(RTRIM(!!string)),
-                        both = sql_expr(LTRIM(RTRIM(!!string))),
-                      )
-                    }
+    sql_expr(REPLACE(!!string, !!pattern, !!replacement))
+  },
+  str_trim = function(string, side = c("both", "left", "right")) {
+    side <- match.arg(side)
+    switch(side,
+      left = sql_expr(LTRIM(!!string)),
+      right = sql_expr(RTRIM(!!string)),
+      both = sql_expr(LTRIM(RTRIM(!!string))),
+    )
+   },
+  str_sub = sql_str_sub("SUBSTR"),
+  str_c = function (..., sep = "", collapse = NULL) sql_not_supported("str_c"),
```

Comment:
I think the not supported functions can just use `...`, since it doesn't actually matter what the arguments are, and it makes the lines a little shorter.

Reply:
Was there a breaking change in `sql_not_supported`? 

I keep getting: `no applicable method for 'escape' applied to an object of class "function"` instead of the "not supported" message

Reply:
I take that back... this may be an implementation problem for `str_detect`...